### PR TITLE
[DOC] detailed installation guide

### PR DESCRIPTION
**Pull request by szha**
## Description ##
add detailed installation guide for installing from master, from source, and dev mode.

### Essentials ###
- [x] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] Code is well-documented

### Changes ###
- [x] add detailed installation guide for installing from master, from source, and dev mode.

### CODE REVIEW

**Comment by szha** [comment]
@eric-haibin-lin @zihaolucky

**Comment by szha** [comment]
@eric-haibin-lin Good point. I will add a hint.

File: docs/index.rst [hunk 1]
```diff
@@ -70,6 +70,13 @@ After installing MXNet, you can install the GluonNLP toolkit by
    pip install gluonnlp
 
 
+.. hint::
+
+   For more detailed guide on installing pre-release from latest master branch,
```
Comment (szha) [inline docs/index.rst]: I didn't call it nightly since we don't actually have a nightly tag. Hope that's ok.

File: docs/install.rst [hunk 2]
```diff
@@ -27,3 +27,50 @@ After installing MXNet, you can install the GluonNLP toolkit by
...
+   You may need to use `sudo` in case you run into permission denied error.
+
+
+Alternatively, you can set up the package with development mode, so that local changes are
+immediately reflected in the installed python package
+
+.. code-block:: console
+
+   python setup.py develop
+
```
Comment (eric-haibin-lin) [inline docs/install.rst]: Shall we also add a link on how to install mxnet nightly if users are asked to upgrade mxnet? https://mxnet.incubator.apache.org/install/index.html?version=master&platform=Linux&language=Python&processor=GPU

Comment (szha) [reply docs/install.rst]: Added

File: docs/install.rst [hunk 3]
```diff
@@ -27,3 +27,54 @@ After installing MXNet, you can install the GluonNLP toolkit by
 
    pip install gluonnlp
 
+
+Install from Master Branch
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you are intersted in trying out features on master branch that hasn't been released yet, you have
```
Comment (astonzhang) [inline docs/install.rst]: intersted
-> 
interested

File: docs/install.rst [hunk 4]
```diff
@@ -27,3 +27,54 @@ After installing MXNet, you can install the GluonNLP toolkit by
...
+the option of installing from master branch directly.
+
+
+Install from Github
```
Comment (astonzhang) [inline docs/install.rst]: Github -> GitHub

File: docs/install.rst [hunk 5]
```diff
@@ -27,3 +27,54 @@ After installing MXNet, you can install the GluonNLP toolkit by
...
++++++++++++++++++++
+
+Use the following command to automatically download and install the current code on master branch
```
Comment (astonzhang) [inline docs/install.rst]: append : at the end of the sentence

File: docs/install.rst [hunk 6]
```diff
@@ -27,3 +27,54 @@ After installing MXNet, you can install the GluonNLP toolkit by
...
+
+.. code-block:: console
+
+   pip install https://github.com/dmlc/gluon-nlp/tarball/master
+
+
+Install from Source Code
+++++++++++++++++++++++++
+
+You can also first check out the code locally using git
```
Comment (astonzhang) [inline docs/install.rst]: git -> Git

append : at the end of the sentence

File: docs/install.rst [hunk 7]
```diff
@@ -27,3 +27,54 @@ After installing MXNet, you can install the GluonNLP toolkit by
...
+
+.. code-block:: console
+
+   git clone https://github.com/dmlc/gluon-nlp
+   cd gluon-nlp
+
+then use the provided `setup.py` to install into site-packages
```
Comment (astonzhang) [inline docs/install.rst]: append : at the end of the sentence

File: docs/install.rst [hunk 8]
```diff
@@ -27,3 +27,54 @@ After installing MXNet, you can install the GluonNLP toolkit by
...
+Alternatively, you can set up the package with development mode, so that local changes are
+immediately reflected in the installed python package
+
+.. code-block:: console
+
+   python setup.py develop
+
+.. note::
+
+   The master branch may rely on mxnet nightly builds which are available on pypi,
```
Comment (astonzhang) [inline docs/install.rst]: mxnet -> MXNet
pypi -> PyPI

**Comment by mli** [comment]
Job PR-547/8 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-547/8/index.html

**Pull request merged**